zerver/models.py

```diff
@@ -967,7 +971,7 @@ def build_message_dict(
                 recipient_type_id
         )
 
-        if recipient_type == Recipient.STREAM:
+        if isinstance(display_recipient, text_type):
```

Comment:
Similar to the above comment, I'd prefer to keep the clarity here of essentially a switch statement on `recipient_type`, since that's what is logically happening here.